ui/src/features/annotator/tools/segment-anything-tool/segment-anything-tool.component.tsx

```diff
@@ -0,0 +1,227 @@
+
+        throttleSetMousePosition({ ...point, positive });
+    };
+
+    const onPointerUp = (event: PointerEvent<SVGSVGElement>) => {
+        if (!ref.current) {
+            return;
+        }
+
+        if (event.button !== 0 && event.button !== 2) {
+            return;
+        }
+
+        if (event.pointerType === 'touch') {
+            return;
+        }
+
+        if (!rightClickMode && isRightButton(event)) {
+            return;
+        }
+
+        // The user must first place a positive point as otherwise we can't show a preview
+        if (rightClickMode && isRightButton(event) && points.length === 0) {
+            return;
+        }
+
+        const point = clampPoint(getRelativePoint(ref.current, { x: event.clientX, y: event.clientY }, zoom.scale));
+
+        // In task chain don't allow the user to place a point outside the ROI
+        if (!pointInRectangle(roi, point)) {
+            return;
+        }
+
+        const positive = isPositivePoint(point, result.shapes, isRightButton(event), rightClickMode);
+
+        const shouldKeepPreviousPoints = interactiveMode === true || isRightButton(event);
+
+        addPoint({ x: point.x, y: point.y, positive }, shouldKeepPreviousPoints);
+    };
+
+    const showPreviewShapes = result.shapes.length === 0 && mousePosition !== undefined;
+    const annotations = (showPreviewShapes ? previewShapes : result.shapes).map((shape, idx): Annotation => {
+        return {
+            shape,
+            labels: [{ id: 'id', color: 'red', name: 'Segment Anything', isPrediction: false }],
+            id: `${idx}`,
+        };
+    });
+
+    if (isLoading) {
+        return <AnnotatorLoading isLoading={isLoading} />;
+    }
+
+    return (
+        <SvgToolCanvas
+            image={image}
+            canvasRef={ref}
+            onPointerMove={handleMouseMove}
+            onPointerUp={onPointerUp}
+            onPointerLeave={() => {
+                throttleSetMousePosition.cancel();
+                setMousePosition(undefined);
+                setPreviewShapes([]);
+            }}
+            style={{
+                cursor: interactiveMode
+                    ? `url("/icons/pencil-plus.svg") 16 16, auto`
+                    : `url("/icons/selection.svg") 8 8, auto`,
+            }}
+        >
+            <MaskAnnotations isEnabled annotations={annotations} width={mediaItem.width} height={mediaItem.height}>
+                <></>
+            </MaskAnnotations>
+
+            {previewShapes.length > 0 &&
+                previewShapes.map((shape, idx) => (
+                    <g
+                        key={idx}
+                        aria-label='Segment anything preview'
+                        {...SELECT_ANNOTATION_STYLES}
+                        strokeWidth={'calc(3px / var(--zoom-scale))'}
+                        fillOpacity={0.0}
+                        className={classes.stroke}
```

Comment:
This shape should include the `classes.animateStroke` class so that the border animates, giving the impression that this is not yet a finalized annotation